### PR TITLE
Add default return to getShared function

### PR DESCRIPTION
**Pull request by hailwood**
Just allows `Inertia::getShared('some-key', 'A default value')` similar to `Session::get('some-key', 'A default value');`

### CODE REVIEW

**Review by claudiodekker — APPROVED** [review]

**Comment by reinink** [comment]
Happy to merge this. Curious though, did you actually have a use-case for this? I mostly have `getShared()` for internal purposes (testing).

**Comment by hailwood** [comment]
Hey @reinink,

Yes so I have a custom facade for Inertia with a few methods for doing flash notifications so I can just do

```php
Inertia::successNotification('User Created!');
return Redirect::back();
```

or

```php
Inertia::successNotification('Welcome to the application!', false);
Inertia::render('Welcome');
```

This is implemented via the following

```php
<?php

namespace App\Facades;

use Inertia\Inertia as InertiaFacade;
use Illuminate\Support\Facades\Session;


class Inertia extends InertiaFacade
{
    public static function flashImmediate($type, $message)
    {
        $flash   = static::getShared('flash-immediate') ?? [];
        $flash[] = [
            'type'    => $type,
            'message' => $message
        ];

        InertiaFacade::share('flash-immediate', $flash);
    }

    public static function flashDelayed($type, $message)
    {
        $flash   = Session::get('flash', []);
        $flash[] = [
            'type'    => $type,
            'message' => $message
        ];

        Session::flash('flash', $flash);
    }

    public static function flash($type, $message, $immediate = false)
    {
        $immediate ? static::flashImmediate($type, $message) : static::flashDelayed($type, $message);
    }

    public static function successNotification($message, $immediate = false)
    {
        static::flash('success', $message, $immediate);
    }

    public static function infoNotification($message, $immediate = false)
    {
        static::flash('info', $message, $immediate);
    }

    public static function warningNotification($message, $immediate = false)
    {
        static::flash('warning', $message, $immediate);
    }

    public static function errorNotification($message, $immediate = false)
    {
        static::flash('error', $message, $immediate);
    }
}
```

And then in the service provider I share a DataProvider (see #118 for details on those)

```php
Inertia::share((new InertiaSharedDataProvider())->toArray());
```

```php
<?php

namespace App\Http\DataProviders;

use App\Models\User;
use App\Facades\Inertia;
use App\Models\Organisation;
use Illuminate\Support\ViewErrorBag;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Session;

class InertiaSharedDataProvider extends DataProvider
{

    public function errors(): array
    {
        return Session::get('errors', new ViewErrorBag)->getBag('default')->getMessages();
    }

    public function flash(): array
    {
        return array_merge(Inertia::getShared('flash-immediate') ?? [], Session::get('flash', []));
    }

   //... other stuff
    
}
```

While I could do this other ways, such as having a custom store for the notifications, or using one of the many available flash notification packages, I figured I might as well just use intertia's built in store.

So long story short, it's just so I can swap out the `static::getShared('flash-immediate') ?? []` to `static::getShared('flash-immediate', [])`

**Comment by reinink** [comment]
So, you're really using this for flash messages...that's cool. Have you seen how I've done this in the [Ping CRM](https://github.com/inertiajs/pingcrm) demo app? I just use Laravel's built-in flash features.

For example:

```php
// Controller
return Redirect::route('users')->with('success', 'User created.');
```

And then I share this with my client-side app like this:

```php
// AppServiceProvider
Inertia::share([
    'flash' => function () {
        return [
            'success' => Session::get('success'),
            'error' => Session::get('error'),
        ];
    },
}
```

**Comment by hailwood** [comment]
I have yes, I use pingcrm as a goto for a lot of things :+1: 

I believe the only usecase it doesn't cover is the immediate notifications which is what I use the shared data for, 
the standard flash notifications do actually go through the session flash data (and so your `->with` example would still work).

```php
Inertia::successNotification('Welcome to the application!', true);
Inertia::render('Welcome');
```

**Comment by claudiodekker** [comment]
@reinink I think we can safely merge this one? Looking at the code, I'm honestly quite surprised we weren't doing this already, as it seems like a well-known convention in Laravel regardless.

**Comment by reinink** [comment]
Released: https://inertiajs.com/releases/inertia-laravel-0.4.1
